ibm/service/power/data_source_ibm_pi_datacenter.go

```diff
@@ -180,7 +185,18 @@ func dataSourceIBMPIDatacenterRead(ctx context.Context, d *schema.ResourceData,
 	if region, ok := d.GetOk(Arg_DatacenterZone); ok {
 		datacenterZone = region.(string)
 	}
-	client := instance.NewIBMPIDatacenterClient(ctx, sess, "")
+
+	var client *instance.IBMPIDatacentersClient
+	if !sess.IsOnPrem() {
+		client = instance.NewIBMPIDatacenterClient(ctx, sess, "")
+	} else {
+		if cloudInstanceId, ok := d.GetOk(Arg_CloudInstanceID); ok {
+			client = instance.NewIBMPIDatacenterClient(ctx, sess, cloudInstanceId.(string))
+		} else {
+			return diag.Errorf("must provide %s if the datacenter is private/on-prem", Arg_CloudInstanceID)
+		}
+	}
+
```

Comment:
Can you update it so the on premise check is done in the sdk and only have

```
var cloudInstance := ""
if cloudInstanceId, ok := d.GetOk(Arg_CloudInstanceID); ok {
			cloudInstance = cloudInstance
		} 
client = instance.NewIBMPIDatacenterClient(ctx, sess, cloudInstance)
```

And update the doc to say the instanceID is required if onprem

Reply:
Fixed above. Docs already specify it as required if private.